temporal-sdk/src/main/java/io/temporal/worker/Worker.java

```diff
@@ -386,11 +400,28 @@ public void registerActivitiesImplementations(Object... activityImplementations)
     workflowWorker.registerLocalActivityImplementations(activityImplementations);
   }
 
+  /**
+   * Register Nexus service implementation objects with a worker.
+   *
+   * <p>An Nexus service object must implement at least one interface annotated with {@link
+   * io.nexusrpc.handler.ServiceImpl}.
+   *
+   * @throws TypeAlreadyRegisteredException if one of the services is already registered
+   */
+  @Experimental
+  public void registerNexusServiceImplementation(Object... nexusServiceImplementations) {
+    Preconditions.checkState(
+        !started.get(),
+        "registerNexusServiceImplementation is not allowed after worker has started");
+    nexusWorker.registerNexusServiceImplementation(nexusServiceImplementations);
+  }
+
   void start() {
     if (!started.compareAndSet(false, true)) {
       return;
     }
     workflowWorker.start();
+    nexusWorker.start();
```

Comment:
To save me from digging too much, can you confirm no resources are used by a Nexus worker if no Nexus service impls are registered?

Reply:
Correct